### PR TITLE
Reimplement UDP endpoint listener with Twisted DatagramProtocol

### PR DESCRIPTION
**Pull request by ichorid**
Greatly speed-ups everything!
See  Tribler/tribler#2548

### CODE REVIEW

**Comment by qstokkink** [comment]
Ok to test

**Comment by qstokkink** [comment]
Ok to test

**Comment by qstokkink** [comment]
Ok to test

File: ipv8/messaging/interfaces/udp/endpoint.py [hunk 1]
```diff
@@ -1,223 +1,59 @@
...
+        self.transport.write(packet, socket_address)
 
     def open(self):
-        """
-        Open this Endpoint for sending and receiving packets.
-        """
         for _ in xrange(10000):
             try:
+                self._listening_port = reactor.listenUDP(self._port, self, self._ip, UDP_MAX_SIZE)
+                #self._listening_port.startListening()
```
Comment (qstokkink) [inline ipv8/messaging/interfaces/udp/endpoint.py]: Why is this commented out code here?

File: ipv8/messaging/interfaces/udp/endpoint.py [hunk 2]
```diff
@@ -1,223 +1,59 @@
...
     def close(self, timeout=10.0):
-        """
-        Close this Endpoint for sending and receiving.
-        :param timeout: the maximum amount of time to wait for the socket to close
-        :return: whether closing occurred naturally
-        """
         self._running = False
         result = True
+        #FIXME: this thing returns Deferred, need to account for that
+        self._listening_port.stopListening()
```
Comment (qstokkink) [inline ipv8/messaging/interfaces/udp/endpoint.py]: Why not just return this deferred and get rid of the timeout parameter?

**Comment by ichorid** [comment]
So I just removed the timeout argument completely. Anyways, it was not used anywhere.

**Pull request merged**